CMakeLists.txt

```diff
@@ -0,0 +1,108 @@
+# Project information.
+cmake_minimum_required( VERSION 3.5.0 )
+project( AwsIotDeviceSdkEmbeddedC
+         VERSION 4.0.2
+         LANGUAGES C )
+
+# Allow the project to be organized into folders.
+set_property( GLOBAL PROPERTY USE_FOLDERS ON )
+
+# Use C90.
+set( CMAKE_C_STANDARD 90 )
+set( CMAKE_C_STANDARD_REQUIRED ON )
+
+# Do not allow in-source build.
+if( ${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR} )
+    message( FATAL_ERROR "In-source build is not allowed. Please build in a separate directory, such as ${PROJECT_SOURCE_DIR}/build." )
+endif()
+
+# Import global configurations.
+include("tools/cmake/filePaths.cmake")
+
+# Configure options to always show in CMake GUI.
+option( BUILD_TESTS
+        "Set this to ON to build both demo and test executables. When OFF, only demo executables are built."
+        ON )
+option( BUILD_CLONE_SUBMODULES
+        "Set this to ON to automatically clone any required Git submodules. When OFF, submodules must be manually cloned."
+        ON )
+
+# Unity test framework does not export the correct symbols for DLLs.
+set( ALLOW_SHARED_LIBRARIES ON )
+    
+include( CMakeDependentOption )
+CMAKE_DEPENDENT_OPTION( BUILD_SHARED_LIBS
+                        "Set this to ON to build all libraries as shared libraries. When OFF, libraries build as static libraries."
+                        ON "${ALLOW_SHARED_LIBRARIES}"
+                        OFF )
+
+# Set the platform named based on the host OS if not defined.
+if( NOT DEFINED PLATFORM_NAME )
```

Comment:
I would suggest keeping platform specific configuration in the test/demos' CMakeLists instead of being present in the master CMakeLists.txt. We want the test/demos to be self-contained with information of platform-specific stack being defined by them.